### DISCUSSION TITLE
wire:dirty not works if disconected

### DISCUSSION THREAD

**Discussion by juliangarcess** [livewire/livewire#8758] [7. Bugs (archived)] [answered]
### Livewire version

v3.5.4

### Laravel version

v10.47.0

### Which PHP version are you using?

PHP 8.1

### Steps To Reproduce

1. Increment/decrement the counter with devtools internet enabled, all works as expected.
2. disconnect the internet and increment/decrement the counter, once the request fail the dirty state gets removed and appears as if is in-sync (but is not).

### Repository or Wirebox URL

https://wirebox.app/b/4ko01

### Additional information

I think the dirty state only can be removed when the component is in-sync with the server and not when the request fails

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [x] I have provided code samples as text and NOT images.
- [x] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by juliangarcess** [comment]
I think this behavior should be fixed but idk how.

**Comment by juliangarcess** [comment]
![image](https://github.com/user-attachments/assets/d287f112-f3d4-4f5d-90f5-84c98c815a81)

Changing this on https://github.com/livewire/livewire/blob/main/js/directives/wire-dirty.js solves the issue.

**Comment by joshhanley** [comment, marked as answer]
Fixed in PR #8956 thanks to @juliangarcess!
